modules/caddyhttp/caddyhttp.go

```diff
@@ -176,14 +177,15 @@ func (app *App) Start() error {
 		}
 
 		for _, lnAddr := range srv.Listen {
-			network, addrs, err := caddy.ParseNetworkAddress(lnAddr)
+			listenAddr, err := caddy.ParseNetworkAddress(lnAddr)
 			if err != nil {
 				return fmt.Errorf("%s: parsing listen address '%s': %v", srvName, lnAddr, err)
 			}
-			for _, addr := range addrs {
-				ln, err := caddy.Listen(network, addr)
+			for i := listenAddr.FromPort; i <= listenAddr.ToPort; i++ {
+				hostport := net.JoinHostPort(listenAddr.Host, fmt.Sprintf("%d", i))
```

Comment:
There should be a method to do this for us.

(Ideally, using this struct will not make our code more complex, i.e. it shouldn't make us do more work.)